portfolio/src/main/client/src/components/sections/comments.ts

```diff
@@ -330,9 +330,35 @@ const Comment = (comment: CommentData | TranslatedCommentData): HTMLElement => {
       }
     });
 
-    btnsEl.append(translateBtn);
+    controlsDiv.append(translateBtn);
   }
 
+  const expandBtn = htmlElement`
+    <button type="button" class="btn-flat btn-inline comment-expander">
+      see more
+    </button>`;
+
+  expandBtn.addEventListener('click', () => {
+    if (readmore.isCollapsed) {
+      readmore.expand();
+      expandBtn.innerText = 'see less';
+    } else {
+      readmore.collapse();
+      expandBtn.innerText = 'see more';
+    }
+  });
+
+  controlsDiv.append(expandBtn);
+
+  // hack because there are no real 'components' here which means
+  // there is no way to know when the component is added to the DOM
+  setTimeout(() => {
```

Comment:
I suggest using a ResizeObserver on the host element:
https://developer.mozilla.org/en-US/docs/Web/API/ResizeObserver

This will fire when the element is added to the dom, and will also update as the element contents change in a way that triggers a resize.

Reply:
Wow, I didn't know that the `ResizeObserver` existed. That must be new. Thanks!

Reply:
https://caniuse.com/#search=ResizeObserver The amount of browser support for `ResizeObserver` doesn't satisfy me, though. Most importantly, it is not supported in Firefox for Android, which I think is an important target for compatibility.

Reply:
I could use `ResizeObserver` when it exists, and otherwise fall back to the hack.

Reply:
lol, I'm surprised Firefox on Android is where the line is drawn. I'd expect this to be very difficult to polyfill given that it's a dom level API, but there's a draft of the spec here:
https://drafts.csswg.org/resize-observer/#resize-observer-interface

You're welcome to just use the setTimeout here. Browser compatibility is a good reason not to use an API.